Assets/MixedRealityToolkit-Preview/CameraCapture/CameraCaptureScreen.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using Unity.Collections;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Preview.CameraCapture
+{
+	public class CameraCaptureScreen : ICameraCapture
+	{
+		private Camera           sourceCamera;
+		private Texture2D        captureTex  = null;
+		private bool             initialized = false;
+		private CameraResolution resolution  = null;
+		private int              renderMask  = ~(1 << 31);
+
+		/// <summary>
+		/// Is the camera completely initialized and ready to begin taking pictures?
+		/// </summary>
+		public bool  IsReady           { get { return initialized; } }
+		/// <summary>
+		/// Is the camera currently already busy with taking a picture?
+		/// </summary>
+		public bool  IsRequestingImage { get { return false;  } }
+		/// <summary>
+		/// Field of View of the camera in degrees. This value is never ready until after 
+		/// initialization, and in many cases, isn't accurate until after a picture has
+		/// been taken. It's best to check this after each picture if you need it.
+		/// </summary>
+		public float FieldOfView       { get { return Camera.main.fieldOfView; } }
+
+		public CameraCaptureScreen(Camera aSourceCamera, int aRenderMask = ~(1 << 31))
+		{
+			sourceCamera = aSourceCamera;
+			renderMask   = aRenderMask;
+		}
+
+		/// <summary>
+		/// Starts up and selects a device's camera, and finds appropriate picture settings
+		/// based on the provided resolution! 
+		/// </summary>
+		/// <param name="preferGPUTexture">Do you prefer GPU textures, or do you prefer a NativeArray of colors? Certain optimizations may be present to take advantage of this preference.</param>
+		/// <param name="resolution">Preferred resolution for taking pictures, note that resolutions are not guaranteed! Refer to CameraResolution for details.</param>
+		/// <param name="onInitialized">When the camera is initialized, this callback is called! Some cameras may return immediately, others may take a while. Can be null.</param>
+		public void Initialize(bool aPreferGPUTexture, CameraResolution aResolution, Action aOnInitialized)
+		{
+			resolution  = aResolution;
+			initialized = true;
+
+			if (aOnInitialized != null)
+				aOnInitialized();
+		}
+
+		private void GrabScreen(Vector2Int aSize)
+		{
+			if (captureTex == null || captureTex.width != aSize.x || captureTex.height != aSize.y)
```

Comment:
doesnt feel like we need to check captureTex == null here... it is checked for != null in the next condition and destroyed.

Reply:
captureTex can be either null or !null in this case. If it's null, it needs created, if it's not null and the dimensions don't match, it needs destroyed and recreated. I use this pattern a fair bit to avoid having two identical or near identical texture creation lines so close together.

If you have a better patterns for this type of case, I'm open to it!